sky/cli.py

```diff
@@ -3471,11 +3474,22 @@ def storage_delete(names: List[str], all: bool):  # pylint: disable=redefined-bu
     if sum([len(names) > 0, all]) != 1:
         raise click.UsageError('Either --all or a name must be specified.')
     if all:
-        click.echo('Deleting all storage objects.')
         storages = sky.storage_ls()
         names = [s['name'] for s in storages]
     else:
         names = _get_glob_storages(names)
+    if names:
+        if not yes:
+            storage_names = ', '.join(names)
+            storage_str = 'storages' if len(names) > 1 else 'storage'
+            click.confirm(
+                f'Deleting {len(names)} {storage_str}: '
+                f'{storage_names}. Proceed?',
+                default=True,
+                abort=True,
+                show_default=True)
+    elif all:
+        click.echo('No storage(s) to delete.')
```

Comment:
@aseriesof-tubes Sorry I missed to mention one last fix. Let's move this check under L3477. We can check if `storages` is empty or not and echo `'No storage(s) to delete.'` and then `return`. When storage names are specified for `sky storage delete`, it's echoed in `_get_glob_storages`, and we want to handle the echo under the same logical block of `if else`.

Reply:
No problem! I agree that would be a cleaner approach. All changes have been made, and it is now ready for another review!